packages/e2e/test/web-extension/extension/background/config.ts

```diff
@@ -75,35 +86,70 @@ const blockfrostApi = [
   : null;
 
 export const utxoProvider = (async () => {
-  if (env.UTXO_PROVIDER === 'blockfrost') {
-    return blockfrostUtxoProvider(await blockfrostApi!);
+  switch (env.UTXO_PROVIDER) {
+    case 'blockfrost': {
+      return blockfrostUtxoProvider(await blockfrostApi!);
+    }
+    case 'http': {
+      return utxoHttpProvider({ adapter: axiosFetchAdapter, baseUrl: env.UTXO_PROVIDER_PARAMS.url, logger });
+    }
+    default: {
+      throw new Error(`UTXO_PROVIDER unsupported: ${env.UTXO_PROVIDER}`);
+    }
   }
-  throw new Error(`UTXO_PROVIDER unsupported: ${env.UTXO_PROVIDER}`);
 })();
 
 export const assetProvider = (async () => {
-  if (env.ASSET_PROVIDER === 'blockfrost') {
-    return blockfrostAssetProvider(await blockfrostApi!);
+  switch (env.ASSET_PROVIDER) {
+    case 'blockfrost': {
+      return blockfrostAssetProvider(await blockfrostApi!);
+    }
+    case 'http': {
+      return assetInfoHttpProvider({ adapter: axiosFetchAdapter, baseUrl: env.ASSET_PROVIDER_PARAMS.url, logger });
+    }
+    default: {
+      throw new Error(`ASSET_PROVIDER unsupported: ${env.ASSET_PROVIDER}`);
+    }
   }
-  throw new Error(`ASSET_PROVIDER unsupported: ${env.ASSET_PROVIDER}`);
 })();
 
 export const txSubmitProvider = (async () => {
   switch (env.TX_SUBMIT_PROVIDER) {
     case 'blockfrost': {
       return blockfrostTxSubmitProvider(await blockfrostApi!);
     }
+    case 'http': {
+      return txSubmitHttpProvider({ adapter: axiosFetchAdapter, baseUrl: env.TX_SUBMIT_PROVIDER_PARAMS.url, logger });
```

Comment:
```suggestion
      return txSubmitHttpProvider({ adapter: axiosFetchAdapter, baseUrl: env.TX_SUBMIT_PROVIDER_PARAMS.baseUrl, logger });
```
Better to be consistent. This will be easier when typing the params ENV, as you can pick properties.

Reply:
addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/23af88930eb60cf2d4c45d5b044142e8c421e62b